## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 360950`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>git-credential-gopass</li>
    <li>gopass</li>
    <li>gopass-hibp</li>
    <li>gopass-summon-provider</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>git-credential-gopass</li>
    <li>gopass</li>
    <li>gopass-hibp</li>
    <li>gopass-summon-provider</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>git-credential-gopass</li>
    <li>gopass</li>
    <li>gopass-hibp</li>
    <li>gopass-summon-provider</li>
  </ul>
</details>

---


Failed in macos

pr-360950/logs/gopass-x86_64-darwin.log

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/6aa55h0gjgqrq5d85r0sp9gx12v0kcx9-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Building subPackage ./.
# github.com/kbinani/screenshot
vendor/github.com/kbinani/screenshot/darwin.go:9:10: fatal error: 'ScreenCaptureKit/ScreenCaptureKit.h' file not found
#include <ScreenCaptureKit/ScreenCaptureKit.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

--- a/pkgs/tools/security/gopass/default.nix
+++ b/pkgs/tools/security/gopass/default.nix
@@ -1,21 +1,25 @@
-{ lib
-, stdenv
-, makeWrapper
-, buildGoModule
-, fetchFromGitHub
-, installShellFiles
-, git
-, gnupg
-, xclip
-, wl-clipboard
-, passAlias ? false
+{
+  lib,
+  stdenv,
+  makeWrapper,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  git,
+  gnupg,
+  xclip,
+  wl-clipboard,
+  passAlias ? false,
 }:
 
 buildGoModule rec {
   pname = "gopass";
   version = "1.15.14";
 
-  nativeBuildInputs = [ installShellFiles makeWrapper ];
+  nativeBuildInputs = [
+    installShellFiles
+    makeWrapper
+  ];
 
   src = fetchFromGitHub {
     owner = "gopasspw";
@@ -28,25 +32,33 @@ buildGoModule rec {
 
   subPackages = [ "." ];
 
-  ldflags = [ "-s" "-w" "-X main.version=${version}" "-X main.commit=${src.rev}" ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.commit=${src.rev}"
+  ];
 
   wrapperPath = lib.makeBinPath (
     [
       git
       gnupg
       xclip
-    ] ++ lib.optional stdenv.hostPlatform.isLinux wl-clipboard
+    ]
+    ++ lib.optional stdenv.hostPlatform.isLinux wl-clipboard
   );
 
-  postInstall = ''
-    installManPage gopass.1
-    installShellCompletion --cmd gopass \
-      --zsh zsh.completion \
-      --bash bash.completion \
-      --fish fish.completion
-  '' + lib.optionalString passAlias ''
-    ln -s $out/bin/gopass $out/bin/pass
-  '';
+  postInstall =
+    ''
+      installManPage gopass.1
+      installShellCompletion --cmd gopass \
+        --zsh zsh.completion \
+        --bash bash.completion \
+        --fish fish.completion
+    ''
+    + lib.optionalString passAlias ''
+      ln -s $out/bin/gopass $out/bin/pass
+    '';
 
   postFixup = ''
     wrapProgram $out/bin/gopass \
@@ -61,7 +73,10 @@ buildGoModule rec {
     description = "Slightly more awesome Standard Unix Password Manager for Teams. Written in Go";
     homepage = "https://www.gopass.pw/";
     license = licenses.mit;
-    maintainers = with maintainers; [ rvolosatovs sikmir ];
+    maintainers = with maintainers; [
+      rvolosatovs
+      sikmir
+    ];
     changelog = "https://github.com/gopasspw/gopass/blob/v${version}/CHANGELOG.md";
 
     longDescription = ''

--- a/pkgs/tools/security/gopass/default.nix
+++ b/pkgs/tools/security/gopass/default.nix
@@ -10,7 +10,9 @@
   xclip,
   wl-clipboard,
   passAlias ? false,
+  testers,
   nix-update-script,
+  gopass,
 }:
 
 buildGoModule rec {
@@ -68,6 +70,10 @@ buildGoModule rec {
   '';
   passthru = {
     inherit wrapperPath;
+
+    tests.version = testers.testVersion {
+      package = gopass;
+    };
 
     updateScript = nix-update-script { };
   };

--- a/pkgs/tools/security/gopass/default.nix
+++ b/pkgs/tools/security/gopass/default.nix
@@ -10,6 +10,7 @@
   xclip,
   wl-clipboard,
   passAlias ? false,
+  nix-update-script,
 }:
 
 buildGoModule rec {
@@ -67,6 +68,8 @@ buildGoModule rec {
   '';
   passthru = {
     inherit wrapperPath;
+
+    updateScript = nix-update-script { };
   };
 
   meta = with lib; {

